### PR TITLE
Triage and enable all possible wasm backend tests

### PR DESCRIPTION
**Pull request by kripken**
I went through all the tests marked as `no_wasm_backend`. A fair bit now work! Those are enabled here.

I clarified some comments on the non-working ones, and filed issues for the small number that seem like they should pass, or at least are in need of looking into, and linked them.

Some tiny test fixes were needed in some cases. I also fixed a tiny bug in `llvm_opt` for the wasm backend, which it turns out is only used in the one place in one test.

### CODE REVIEW

File: tests/test_browser.py [hunk 1]
```diff
@@ -187,10 +186,9 @@ def test_zzz_html_source_map(self):
 
   @no_wasm_backend('wasm source maps')
   def test_emscripten_log(self):
-    # TODO: wasm support for source maps
+    # TODO: wasm support for source maps. emscripten_loadSourceMap looks at $HTML.map but it should be $NAME.wasm.map.
```
Comment (quantum5) [inline tests/test_browser.py]: Is `emscripten_load` supposed to be here?

File: tests/test_core.py [hunk 2]
```diff
@@ -1974,13 +1974,11 @@ def test_biggerswitch(self):
 59899: 598995989959899
 Success!''')
 
-  @no_wasm_backend('no implementation of computed gotos')
```
Comment (sbc100) [inline tests/test_core.py]: Isn't this still unavailable?

File: tests/test_other.py [hunk 3]
```diff
@@ -3286,7 +3283,7 @@ def test_precompiled_headers(self):
       output = run_js('a.out.js')
       assert '|5|' in output, output
 
-  @no_wasm_backend()
+  @no_wasm_backend('extra fastcomp warnings on unaligned loads/stores, which matter a lot more in asm.js')
```
Comment (sbc100) [inline tests/test_other.py]: "tests extra..."

**Comment by VirtualTim** [comment]
Is there a list of features that don't currently work with the WASM backend?

**Comment by sbc100** [comment]
On Sat, Jul 20, 2019 at 7:56 PM Tim Lander <notifications@github.com> wrote:

> Is there a list of features that don't currently work with the WASM
> backend?
>

We've been using the bugtracker to keep track of the list:
https://github.com/emscripten-core/emscripten/projects/1

I think we are very close to zero missing features now.  Still some bugs of
course, but nothing major that I know of.

File: tests/test_browser.py (hunk 1 above)
Comment (kripken) [reply tests/test_browser.py]: Yes, `emscripten_loadSourceMap` is an internal API that has the problem described here.

File: tests/test_core.py (hunk 2 above)
Comment (kripken) [reply tests/test_core.py]: No, @dscuff fixed things in https://bugs.llvm.org/show_bug.cgi?id=42498

**Comment by kripken** [comment]
Yes, I think there are basically no missing features at this point, except perhaps for the minor things mentioned in this PR, like stack overflow assertions.

**Comment by VirtualTim** [comment]
Thanks @sbc100. So does that mean the fetch API is working? I think that was the only thing I was waiting on.

**Comment by kripken** [comment]
@VirtualTim there is still https://github.com/emscripten-core/emscripten/issues/7024 - basic fetch all works, but not the pthreads extension with the fetch-worker.

**Pull request merged**